src/main.ts

```diff
@@ -24,26 +36,42 @@ const hasButtonClass = "has-run-code-button";
 
 export default class ExecuteCodePlugin extends Plugin {
 	settings: ExecutorSettings;
+	executors: ExecutorContainer;
 
 	/**
 	 * Preparations for the plugin (adding buttons, html elements and event listeners).
 	 */
 	async onload() {
 		await this.loadSettings();
 		this.addSettingTab(new SettingsTab(this.app, this));
+		
+		this.executors = new ExecutorContainer(this);
 
-		this.addRunButtons(document.body);
+		this.iterateOpenFilesAndAddRunButtons();
 		this.registerMarkdownPostProcessor((element, _context) => {
-			this.addRunButtons(element);
+			this.addRunButtons(element, _context.sourcePath);
 		});
 
 		// live preview renderers
 		supportedLanguages.forEach(l => {
 			console.debug(`Registering renderer for ${l}.`)
-			this.registerMarkdownCodeBlockProcessor(`run-${l}`, async (src, el, _ctx) => {
```

Comment:
Language prefixes apart from the run- prefix were removed in the latest development builds, can this be reverted?

Reply:
Sure! 

Should the `languagePrefixes` array be removed too? Would that be in the scope of this PR?

Thank you for reviewing :)

Reply:
I think it can be removed for now, I don't think it was used for anything else. Thanks for this PR btw.

Reply:
Done; thanks!